yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -826,12 +849,174 @@ func (p *ParserIssueDetector) IsUnloggedTablesIssueFiltered() bool {
+func (p *ParserIssueDetector) hasIndexCoverage(index *queryparser.Index, fkColumns []string) bool {
+	// Extract non-expression columns from the index
+	indexColumns := make([]string, 0)
+	for _, param := range index.Params {
+		if !param.IsExpression {
+			indexColumns = append(indexColumns, param.ColName)
+		}
+	}
+
+	// Check if we have enough columns in the index
+	if len(indexColumns) < len(fkColumns) {
+		return false
+	}
+
+	// Check if the FK columns are a subset of the first len(fkColumns) index columns
+	// This handles both exact order and permutation cases
+	indexPrefix := indexColumns[:len(fkColumns)]
+	return utils.IsSetEqual(fkColumns, indexPrefix)
+}
+
+// createMissingFKIndexIssue creates a QueryIssue from a foreign key constraint
+func (p *ParserIssueDetector) createMissingFKIndexIssue(fk ForeignKeyConstraint) QueryIssue {
+	return NewMissingForeignKeyIndexIssue(
+		"TABLE",
+		fk.TableName,
+		"", // sqlStatement - we don't have this in stored constraint
+		strings.Join(fk.ColumnNames, ", "),
+		fk.TableName,
+		fk.ReferencedTable,
+	)
+}
+
 func (p *ParserIssueDetector) SetColumnStatistics(columnStats []utils.ColumnStatistics) {
 	for _, stat := range columnStats {
 		p.columnStatistics[stat.GetQualifiedColumnName()] = stat
 	}
 }
 
+// addIndexToCoverage adds an index to the table indexes map for checking missing foreign key index issue
+func (p *ParserIssueDetector) addIndexToCoverage(index *queryparser.Index) {
+	tableName := index.GetTableName()
+
+	// Add the index to the table's index list
+	if _, exists := p.tableIndexes[tableName]; !exists {
+		p.tableIndexes[tableName] = make([]*queryparser.Index, 0)
+	}
+	p.tableIndexes[tableName] = append(p.tableIndexes[tableName], index)
+}
+
+// addConstraintAsIndex creates a mock index object from a constraint and adds it to the table indexes map.
+// This is used for primary keys and unique constraints which are also indexes for missing foreign key index detection.
+func (p *ParserIssueDetector) addConstraintAsIndex(tableName string, columns []string, constraintType string) {
+	// Create mock index parameters from columns
+	indexParams := make([]queryparser.IndexParam, len(columns))
+	for i, colName := range columns {
+		indexParams[i] = queryparser.IndexParam{
+			ColName:      colName,
+			IsExpression: false,
+		}
+	}
+
+	// Extract schema name from tableName (format: "schema.table")
+	schemaName := ""
+	tableNameOnly := tableName
+	if strings.Contains(tableName, ".") {
+		parts := strings.SplitN(tableName, ".", 2)
+		schemaName = parts[0]
+		tableNameOnly = parts[1]
+	}
+
+	// Create a mock index object
+	mockIndex := &queryparser.Index{
+		SchemaName:   schemaName,
+		IndexName:    fmt.Sprintf("%s_%s", strings.ToLower(constraintType), strings.Join(columns, "_")),
+		TableName:    tableNameOnly,
+		AccessMethod: BTREE_ACCESS_METHOD, // Primary keys and unique constraints use btree by default
```

Comment:
i don't think this is true. You can have other index types as well in a PK. Ideally, you should get the access method from the constraint type.

Reply:
Since it is mentioned in the docs that by default a btree is created, keeping this